plugins/inputs/vsphere/vsan.go

```diff
@@ -0,0 +1,532 @@
+package vsphere
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	vsanmethods "github.com/vmware/govmomi/vsan/methods"
+	vsantypes "github.com/vmware/govmomi/vsan/types"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+)
+
+const (
+	vsanNamespace    = "vsan"
+	vsanPath         = "/vsanHealth"
+	hwMarksKeyPrefix = "vsan-perf-"
+	perfPrefix       = "performance."
+)
+
+var (
+	vsanPerfMetricsName    string
+	vsanSummaryMetricsName string
+	perfManagerRef         = types.ManagedObjectReference{
+		Type:  "VsanPerformanceManager",
+		Value: "vsan-performance-manager",
+	}
+	hyphenReplacer = strings.NewReplacer("-", "")
+)
+
+// collectVsan is the entry point for vsan metrics collection
+func (e *Endpoint) collectVsan(ctx context.Context, resourceType string, acc telegraf.Accumulator) error {
+	lower, err := versionLowerThan(e.apiVersion, "5.5")
+	if err != nil {
+		return fmt.Errorf("failed to get the vCenter version for vSAN: %v", err)
+	}
+	if lower {
+		return fmt.Errorf("a minimum API version of 5.5 is required for vSAN. Found: %s. Skipping vCenter: %s", e.apiVersion, e.URL.Host)
+	}
```

Comment:
Please use github.com/coreos/go-semver/semver for this comparison instead of implementing your own...
```suggestion
	version, err := semver.NewVersion(e.apiVersion)
	if err != nil {
		return fmt.Errorf("failed to get the vCenter version for vSAN: %v", err)
	}
	if version.LessThan(semver.Version{Major: 5, Minor: 5}) 
		return fmt.Errorf("a minimum API version of 5.5 is required for vSAN. Found: %s. Skipping vCenter: %s", e.apiVersion, e.URL.Host)
	}
```